containers/src/Kokkos_Bitset.hpp

```diff
@@ -394,11 +394,12 @@ void deep_copy(Bitset<DstDevice>& dst, Bitset<SrcDevice> const& src) {
         "Error: Cannot deep_copy bitsets of different sizes!");
   }
 
-  using raw_deep_copy =
-      Kokkos::Impl::DeepCopy<typename DstDevice::memory_space,
-                             typename SrcDevice::memory_space>;
-  raw_deep_copy(dst.m_blocks.data(), src.m_blocks.data(),
-                sizeof(unsigned) * src.m_blocks.extent(0));
+  Kokkos::fence("Bitset::deep_copy: fence before copy operation");
```

Comment:
Would be neat, though not blocking, to make any of these DeepCopy fence's report the space names

Reply:
Oh, for all the fences I introduced we can consider if we want to just fence are particular space instead of using the sledgehammer `Kokkos::fence()`.
I just want to check at the moment if anything breaks if we don't fence in any case in `DeepCopy` introducing the fences I think are missing in the call site.

Reply:
I just mean it would be nice for Profiling if we say "Bitset::deep_copy<HostSpace,CudaSpace>: fence before copy operation"

Reply:
Oh, sure.

Reply:
Let me just do that if the `HostSpace` is involved. It doesn't seem all that helpful to reference random spaces.